test/tools/transformer/template_cache_generator_spec.dart

```diff
@@ -0,0 +1,256 @@
+
+                @NgTemplateCache(preCacheUrls: ["test1.html"])
+                class A {}
+
+                main() {}
+            ''',
+            'a|web/dir/foo.dart': '''
+                import 'package:angular/angular.dart';
+                import 'package:angular/angular_cache.dart';
+
+                @NgTemplateCache(preCacheUrls: ["test2.html"])
+                class B {}
+            ''',
+            'a|test1.html': htmlContent1,
+            'a|test2.html': htmlContent2,
+            'angular|lib/angular.dart': libAngular,
+            'angular|lib/angular_cache.dart': libCacheAnnotation,
+          },
+          cacheContent: {
+            'test1.html': htmlContent1,
+            'test2.html': htmlContent2,
+          });
+    });
+
+    it('should handle no cached values', () {
+      return generates(setupPhases(),
+          inputs: {
+              'a|web/main.dart': '''
+                    import 'package:angular/angular.dart';
+                    import 'package:angular/template_cache.dart';
+
+                    main() {}
+                    ''',
+              'angular|lib/angular.dart': libAngular,
+              'angular|lib/template_cache.dart': libCacheAnnotation,
+          });
+    });
+
+    it('should warn on no angular imports', () {
+      return generates(setupPhases(),
+          inputs: {
+              'a|web/main.dart': '''
+                    main() {}
+                    ''',
+              'angular|lib/angular.dart': libAngular,
+          },
+          messages: [
+              'warning: Unable to resolve '
+              'angular.core.annotation_src.Component.'
+          ]);
+      });
+
+    it('should not generate template cache', () {
+      return generates(setupPhases(generateTemplateCache: false),
+          cacheGenerated: false,
+          inputs: {
+            'a|web/main.dart': '''
+                import 'package:angular/angular.dart';
+                import 'dir/foo.dart';
+
+                @Component(templateUrl: "test1.html", cssUrl: "test1.css")
+                class A {}
+
+                main() {}
+            ''',
+            'a|web/dir/foo.dart': '''
+                import 'package:angular/angular.dart';
+
+                @NgTemplateCache(preCacheUrls: ["test2.html"])
+                class B {}
+            ''',
+          'angular|lib/angular.dart': libAngular,
+          'angular|lib/template_cache.dart': libCacheAnnotation,
+      });
+    });
+  });
+}
+
+Future generates(List<List<Transformer>> phases,
+                 { Map<String, String> inputs,
+                 Map<String, String> cacheContent: const {},
+                 Iterable<String> messages: const [],
+                 bool cacheGenerated: true}) {
+
+  var buffer = new StringBuffer();
+  buffer.write(header);
+  cacheContent.forEach((uri, contents) {
+    contents = contents.replaceAll("'''", r"\'\'\'");
+    buffer.write("  r'$uri' : r'''$contents''',\n");
```

Comment:
Added the new line as suggested. I'll need to think some more about a different type of test. It isn't that simple. Maybe a full end-to-end test like relative type urls would work. Can you do that in a separate pull request?